ExampleMod/Content/Items/Weapons/ExampleYoyo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using ExampleMod.Content.Projectiles;
+using ExampleMod.Content.Tiles.Furniture;
+using ExampleMod.Content.Rarities;
+using Terraria;
+using Terraria.ID;
+using Terraria.GameContent.Creative;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Items.Weapons
+{
+	public class ExampleYoyo : ModItem
+	{
+		public override void SetStaticDefaults() {
+			Tooltip.SetDefault("This is a modded yoyo.");
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = 1; //The amount of this item that needs to be researched to unlock it in the journey mode duplication menu.
+
+			// These are all related to gamepad controls and don't seem to affect anything else
+			ItemID.Sets.Yoyo[Item.type] = true;
+			ItemID.Sets.GamepadExtraRange[Item.type] = 15;
+			ItemID.Sets.GamepadSmartQuickReach[Item.type] = true;

```

Comment:
Actual explanations for what these do would be useful. For example, what does `GamepadExtraRange` actually mean? What "range" does it extend?